### PR TITLE
RocksDB Cmake changes for Arm64 CRC32 Optimization

### PR DESCRIPTION
**Pull request by guyuqi**
Add CMake build for RocksDB CRC32 Optimization on Arm64.

### CODE REVIEW

**Comment by miasantreble** [comment]
thanks for your contribution!
like last time, we still don't have a platform to easily test the changes. 
Can you please run crc32_test on your platform and make sure it passes?

File: CMakeLists.txt [hunk 1]
```diff
@@ -201,6 +201,15 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
   endif(HAS_ALTIVEC)
 endif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le")
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
+        CHECK_C_COMPILER_FLAG("-march=armv8-a+crc" HAS_ARMV8_CRC)
+  if(HAS_ARMV8_CRC)
+    message(STATUS " HAS_ARMV8_CRC yes")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crc -Wno-unused-function")
```
Comment (miasantreble) [inline CMakeLists.txt]: is there any particular reason for allowing unused functions?

Comment (guyuqi) [reply CMakeLists.txt]: Really thanks for your comments.
when 'HAS_ARMV8_CRC' is enabled, 'rocksdb::crc32c::isPCLMULQDQ() / isSSE42() ' will not be used but have been actually defined. So it's needed to suppress the warning here.

**Comment by guyuqi** [comment]
> thanks for your contribution!
> like last time, we still don't have a platform to easily test the changes.
> Can you please run crc32_test on your platform and make sure it passes?

Crc32 test result:
```
builder@vm-rocksdb-yq:~/rocksdb/bld$ ls -al crc32c_test
-rwxr-xr-x 1 builder builder 1071480 May 15 05:32 crc32c_test
builder@vm-rocksdb-yq:~/rocksdb/bld$ arch
aarch64
builder@vm-rocksdb-yq:~/rocksdb/bld$ ./crc32c_test
[==========] Running 4 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 4 tests from CRC
[ RUN      ] CRC.StandardResults
[       OK ] CRC.StandardResults (10 ms)
[ RUN      ] CRC.Values
[       OK ] CRC.Values (0 ms)
[ RUN      ] CRC.Extend
[       OK ] CRC.Extend (0 ms)
[ RUN      ] CRC.Mask
[       OK ] CRC.Mask (0 ms)
[----------] 4 tests from CRC (10 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test case ran. (11 ms total)
[  PASSED  ] 4 tests.

```

**Pull request closed without merging**